First implementation of cancel, save und submit feedback + #18

botrufus opened the pull request:
This PR includes:

- refactoring code editor view model to environment object
- cancel, save and submit of assessment
- redesign of general feedback area made by @katjanakosic 

As this is the first implementation of the cancel, save and submit of an assessment. Features that we discussed today are not yet implemented and will be implemented in the future. 
Also the structure (regarding the environment objects) will be redone by me and @pal03377 in another PR. Because of this don't have to be too thoroughly as most changes are highly coupled to the current structure and will be changed anyways.

crustypluto19 commented on Themis.xcodeproj/xcshareddata/xcschemes/Themis.xcscheme:
```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This file shouldn't have been committed as the password of the test server account is exposed again :) 
@huber-florian it's time to get to work again 🙈

huber-florian replied:
No, as far as I see it is not further on develop. So please take care to rebase on develop and squash the commits so that it is also not further part of this feature branch.

botrufus replied:
Its still in the same commit that you deleted @huber-florian today, but as the brach is older it still has the old commits...

crustypluto19 reviewed (commented):
Looks good!

botrufus committed:
cancel, submit and save assessment + redesign of feedbackarea

commit 106293888f1a27afa20ba317de19a53cf89d50b4
Author: botrufus <64660684+botrufus@users.noreply.github.com>
Date:   Wed Nov 30 22:32:39 2022 +0100

    remove addfeedbackview and fix swiftlint error

commit 3d22c6935f4a340b1408cc074c99826c08b2e201
Merge: e849915 35ea657
Author: botrufus <64660684+botrufus@users.noreply.github.com>
Date:   Wed Nov 30 22:29:26 2022 +0100

    merge with #18

commit 35ea657ebc251eb46accab91087565a9a7930203
Author: Evan Christopher <evanchristopher1907@gmail.com>
Date:   Wed Nov 30 16:48:08 2022 +0100

    style changes

commit 979b5bcf2c83bbbbb7368c5a60060e5dc7e66675
Author: Katjana Kosic <ge56yuw@tum.de>
Date:   Wed Nov 30 15:30:55 2022 +0100

    Fix list view and edited feedback being moved to the bottom of the list

commit e8499159a025ffb3f31357185fdd8c1320085ead
Author: botrufus <64660684+botrufus@users.noreply.github.com>
Date:   Wed Nov 30 10:07:29 2022 +0100

    add submit and cancel of assessment

commit 5a1d197ba184481e4b849b6f03755d504684eb8e
Author: botrufus <64660684+botrufus@users.noreply.github.com>
Date:   Wed Nov 30 00:57:14 2022 +0100

    delete old feedback model

commit 9465915480d2646c52d277bbcd3f5dc254c3299b
Author: botrufus <64660684+botrufus@users.noreply.github.com>
Date:   Wed Nov 30 00:55:51 2022 +0100

    delete feedback view model

commit 91ae4d3bc0d6d86e1d10f0286e09803672fbda42
Author: botrufus <64660684+botrufus@users.noreply.github.com>
Date:   Wed Nov 30 00:50:25 2022 +0100

    cancel assessment, make code editor view model environment object, create feedback model for assessment

commit e0ed61454ae55784ef9033aca6afda6b0bc528f4
Author: Alexander Lange <79016456+AlexExcellence@users.noreply.github.com>
Date:   Tue Nov 29 20:21:17 2022 +0100

    Add dynamic syntax highlighting (#19)

commit e233999e930d4d91b8fddf257bfba64f0e882cd7
Author: Katjana Kosic <ge56yuw@tum.de>
Date:   Tue Nov 29 18:16:44 2022 +0100

    Add edit button functionality and redesign for general feedback tab

commit e2e755ff6119ed6776aa75780a19d2612874970e
Author: Florian Huber <116847304+huber-florian@users.noreply.github.com>
Date:   Tue Nov 29 16:57:35 2022 +0100

    Feature/well formatted problem statement (#14)

    Co-authored-by: botrufus <64660684+botrufus@users.noreply.github.com>

commit 5a38aac9326d5add95ae4dc0ecf90f760b0d54f9
Author: Alexander Lange <79016456+AlexExcellence@users.noreply.github.com>
Date:   Tue Nov 29 15:13:39 2022 +0100

    Add Inline Feedbacks (#15)

botrufus committed:
add lost feedbackhighlighting array

pal03377 reviewed (approved):
Works!

crustypluto19 commented on Themis/Views/Assessment/CorrectionSidebar/GeneralFeedbackCellView.swift:
```diff
@@ -8,46 +8,51 @@
-                Section(header: Text("General")) {
-                    ForEach(feedbackModel.generalFeedbacks) { feedback in
-                        FeedbackCellView(feedbackModel: feedbackModel, feedbackID: feedback.id)
+                Section(header: HStack {
```
Right now general and inline feedbacks are saved in the same section, this will be changed in the future :)

crustypluto19 reviewed (approved):
Looks good, code looks cleaner now 👍
